node/app/application.go

```diff
@@ -274,9 +274,17 @@ func (app Application) DeliverTx(tx []byte) ResponseDeliverTx {
 
 	log.Debug("Starting processing")
 	if result.ShouldProcess(app) {
-		ttype, _ := action.UnpackMessage(action.Message(tx))
+		transaction, _ := action.Parse(action.Message(tx))
+
+		// TODO: can we improve this?
+		swapProcessing := false
+		switch transaction.(type) {
+		case *action.Swap:    swapProcessing = true
+		case *action.Publish: swapProcessing = true
+		case *action.Verify:  swapProcessing = true
```

Comment:
golang will let you put all 3 in the same case statement, separated by commas...